src/libos/src/process/spawn_attribute.rs

```diff
@@ -0,0 +1,97 @@
+use super::*;
+use crate::signal::{sigset_t, SigSet};
+use crate::util::mem_util::from_user::check_ptr;
+
+// Note: Only support POSIX_SPAWN_SETSIGDEF and POSIX_SPAWN_SETSIGMASK
+#[derive(Default, Debug, Copy, Clone)]
+pub struct SpawnAttr {
+    pub sig_mask: Option<SigSet>,
+    pub sig_default: Option<SigSet>,
+}
+
+// Glibc and musl use 128 bytes to represent sig_set_t
+type SpawnAttrSigSet = [sigset_t; 16];
+
+// Note: The name of the element follows the glibc style. The comments show the
+// name in musl.
+#[repr(C)]
+#[derive(Debug)]
+pub struct posix_spawnattr_t {
+    flags: SpawnAttributeFlags, // __flags
+    pgrp: i32,                  // __pgrp
+    sd: SpawnAttrSigSet,        // __def
+    ss: SpawnAttrSigSet,        // __mask
+    // Below elements are different in musl but we are not using it for now. So just ignore.
+    sp: SchedParam,
```

Comment:
If we do not use entries, should we add them now? Or can we find a better way to add both glibc and musl implementation?

Reply:
Remove the elements we don't use for now. Also, adjust the definition order to be clearer.